sympy/polys/tests/test_densetools.py

```diff
@@ -637,6 +641,25 @@ def test_dup_clear_denoms():
     assert dup_clear_denoms([EX(7)], EX) == (EX(1), [EX(7)])
     assert dup_clear_denoms([EX(sin(x)/x), EX(0)], EX) == (EX(x), [EX(sin(x)), EX(0)])
 
+    # This test is for inexact domain handling
+    R, x = ring("x", RR)
+    f = RR(0.5)*x + RR(0.333)
+    common, cleared_f = dup_clear_denoms(f, RR)
+    assert cleared_f == f, "Function should not modify input for inexact domains when no clear action is needed"
```

Comment:
Look at the rest of the code in this file for examples of how to write the test code.